src/rules/no-restricted-paths.js

```diff
@@ -56,78 +56,125 @@ module.exports = {
     const basePath = options.basePath || process.cwd();
     const currentFilename = context.getPhysicalFilename ? context.getPhysicalFilename() : context.getFilename();
     const matchingZones = restrictedPaths.filter((zone) => {
-      const targetPath = path.resolve(basePath, zone.target);
+      const allTargets = zone.target instanceof Array ? zone.target : [zone.target];
+      const allTargetPaths = allTargets.map(target => path.resolve(basePath, target));
+      return allTargetPaths.some(targetPath => isMatchingTargetPath(currentFilename, targetPath));
+    });
 
+    function isMatchingTargetPath(filename, targetPath) {
       if (isGlob(targetPath)) {
-        return minimatch(currentFilename, targetPath);
+        return minimatch(filename, targetPath);
       }
 
-      return containsPath(currentFilename, targetPath);
-    });
+      return containsPath(filename, targetPath);
+    }
 
     function isValidExceptionPath(absoluteFromPath, absoluteExceptionPath) {
       const relativeExceptionPath = path.relative(absoluteFromPath, absoluteExceptionPath);
 
       return importType(relativeExceptionPath, context) !== 'parent';
     }
 
+    function areBothGlobPatternAndAbsolutePath(areGlobPatterns) {
+      return areGlobPatterns.some((isGlob) => isGlob) && areGlobPatterns.some((isGlob) => !isGlob);
```

Comment:
these conditions can't ever both be true.

Reply:
@ljharb they can, for example with `from: ["./*", "./"]`
- `areGlobPatterns` will be an array with values `[true, false]` (the first one is a glob pattern, the second one isn't)
- that's exactly the situation I want to avoid, to keep the homogeneity between `from` and `except` as it was before

do you agree with this with this context?

Reply:
ah, i see what you mean, thanks.